daemon/cmd/daemon.go

```diff
@@ -1159,6 +1168,43 @@ func (d *Daemon) bootstrapClusterMesh(nodeMngr *nodemanager.Manager) {
 	bootstrapStats.clusterMeshInit.End(true)
 }
 
+// ReloadOnDeviceChange regenerates device related information and reloads the datapath.
+func (d *Daemon) ReloadOnDeviceChange(devices []string) {
+	log.WithField(logfields.Devices, devices).Infof("Datapath reload initiated due to changed devices.")
+
+	option.Config.Devices = devices
```

Comment:
This makes me a little bit nervous, just because my understanding of the option package today is that there is zero locking protections, we only ever write it on startup, then reuse it all over the place. This is introducing a write operation arbitrarily later based on device changes.

Reply:
Yep, I really want to get rid of this, either by switching everything to use `deviceManager.GetDevices()`, or preferably making the whole thing event based. Would you be OK with this for now, with this feature being behind an experimental flag and a having a follow up soon that does away with this? It's a bigger refactoring and I'd like to do it interleaved with the agent modularization as that makes the whole thing easier.

Reply:
I agree that's the path forward.

Any chance there's a reasonable short-term fix with an atomic pointer or something?